### PR TITLE
refactor(common): remove `supportScrollRestoration`

### PR DESCRIPTION
**Pull request by JeanMeche**
`scrollRestoration` is supported across all everygreen browsers. Let's just keep `supportsScrolling` for the server platform. 

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [x] No

### CODE REVIEW

**Comment by AndrewKushnir** [comment]
@atscott could you please help with the review to see if it's safe to land?

File: packages/common/src/viewport_scroller.ts [hunk 1]
```diff
@@ -160,43 +145,6 @@ export class BrowserViewportScroller implements ViewportScroller {
...
-
-  /**
-   * We only support scroll restoration when we can get a hold of window.
-   * This means that we do not support this behavior when running in a web worker.
-   *
-   * Lifting this restriction right now would require more changes in the dom adapter.
-   * Since webworkers aren't widely used, we will lift it once RouterScroller is
-   * battle-tested.
-   */
-  private supportScrollRestoration(): boolean {
```
Comment (AndrewKushnir) [inline packages/common/src/viewport_scroller.ts:172]: @JeanMeche we've discussed this change internally and we'd like to propose dropping this function, but still keeping the `supportsScrolling` one. This would allow us to make sure the behavior doesn't break in environments where `window` is not available (SSR cases). We do override this class in [the `platform-server` package](https://github.com/angular/angular/blob/main/packages/platform-server/src/server.ts#L53), but we want to make sure we retain the current behavior (don't break when there is no `window`).

File: packages/common/src/viewport_scroller.ts [hunk 2]
```diff
@@ -139,12 +129,7 @@ export class BrowserViewportScroller implements ViewportScroller {
    * Disables automatic scroll restoration provided by the browser.
    */
   setHistoryScrollRestoration(scrollRestoration: 'auto'|'manual'): void {
-    if (this.supportScrollRestoration()) {
-      const history = this.window.history;
-      if (history && history.scrollRestoration) {
-        history.scrollRestoration = scrollRestoration;
-      }
-    }
+    this.window.history.scrollRestoration = scrollRestoration;
```
Comment (AndrewKushnir) [inline packages/common/src/viewport_scroller.ts:132]:
```suggestion
    if (this.supportsScrolling()) {
      this.window.history.scrollRestoration = scrollRestoration;
    }
```

Comment (AndrewKushnir) [reply packages/common/src/viewport_scroller.ts:132]: We can drop the `history.scrollRestoration` check, but retain the `this.supportsScrolling()` one.

File: packages/common/src/viewport_scroller.ts [hunk 3]
```diff
@@ -117,10 +111,6 @@ export class BrowserViewportScroller implements ViewportScroller {
    * @see https://html.spec.whatwg.org/#scroll-to-fragid
    */
   scrollToAnchor(target: string): void {
-    if (!this.supportsScrolling()) {
```
Comment (AndrewKushnir) [inline packages/common/src/viewport_scroller.ts:120]: Let's restore this check as well (see another comment below for context).

File: packages/common/src/viewport_scroller.ts [hunk 4]
```diff
@@ -88,21 +88,15 @@ export class BrowserViewportScroller implements ViewportScroller {
    * @returns The position in screen coordinates.
    */
   getScrollPosition(): [number, number] {
-    if (this.supportsScrolling()) {
```
Comment (AndrewKushnir) [inline packages/common/src/viewport_scroller.ts:91]: I think we'd need to revert the changes in this function and in the `scrollToPosition` as well, so that this becomes a noop (without throwing an error) is window is not available.

Comment (JeanMeche) [reply packages/common/src/viewport_scroller.ts:91]: What you think about this little refactoring to express that `window` can in fact be `undefined` ?

Comment (AndrewKushnir) [reply packages/common/src/viewport_scroller.ts:91]: I think it's ok, but I'd propose moving this to a separate PR, so that:

- this PR focuses on removing no longer needed code (with minimal required refactoring)
- a followup PR would introduce extra cleanup

In this case it'd be easier to review and land the change.

Comment (JeanMeche) [reply packages/common/src/viewport_scroller.ts]: done deal ✅

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[TGP](https://test.corp.google.com/OCL:545853036:BASE:545852629:1688612896303:3594c7cb).

**Comment by AndrewKushnir** [comment]
**Caretaker note**: TGP is "green", this PR is ready for merge.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit b8107b19149a51cbe9ae04bd9e5b829782c7c0f6.
